### DISCUSSION TITLE
Failed to fetch log file from worker

### DISCUSSION THREAD

**Discussion by Dissolution6018** [apache/airflow#29269] [Q&A]
Hello!

**Apache Airflow version:
Airflow version: 2.5.1**

Airflow Web UI is not able to fetch logs (actually stopped working) from the worker's container. It fails with the following error:
```
*** Log file does not exist: /opt/airflow/logs/dag_id=test_dag_1/run_id=scheduled__2023-01-30T21:00:00+00:00/task_id=task/attempt=1.log
*** Fetching from: http://example_host:8793/log/dag_id=test_dag_1/run_id=scheduled__2023-01-30T21:00:00+00:00/task_id=task/attempt=1.log
*** !!!! Please make sure that all your Airflow components (e.g. schedulers, webservers and workers) have the same 'secret_key' configured in 'webserver' section and time is synchronized on all your machines (for example with ntpd) !!!!!
****** See more at https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#secret-key
****** Failed to fetch log file from worker. Client error '403 FORBIDDEN' for url 'http://example_host:8793/log/dag_id=test_dag_1/run_id=scheduled__2023-01-30T21:00:00+00:00/task_id=task/attempt=1.log'
For more information check: https://httpstatuses.com/403
```

The real hostname has been replaced with example_host.

I use single host containers for **airflow**, **celery**, **flower**, **postgresql**, **redis**.
I am upgrading Airflow from version **2.2.2** to **2.5.1**. On **2.2.2** everything works correctly.
After I deploy a new version of airflow and run airflow db init, run a test DAG - this error occurs.

I have tried all methods from https://github.com/apache/airflow/discussions/17251.

All containers use the same airflow.cfg configuration, the same secret_key is set.
There can be no time discrepancies because all containers run on the same machine (host).
I tried changing the secret_key, specifying it via `AIRFLOW_WEBSERVER_SECRET_KEY` and then restarting the scheduler, web server and worker.
Did not work.

A key of the form is used:` jOXX2Cmi2uJow9xRm+Dqag==`
The key was generated via python code:
```
from base64 import b64encode
import os
SECRET_KEY = b64encode(os.urandom(16)).decode('utf-8')
print(SECRET_KEY)
```

Tell me please, how can this be that it works on **2.2.2**, but on **2.5.1** I encounter a problem?

**Comment by potiuk** [comment]
What do the worker logs tell you ? There should be some warning logs explaining why you have forbidden message in the worker logs.  Can you please find them and see what they say ?

**Reply by potiuk** [reply]
And in case you have to little information you can anable debug log (https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#logging-level) and find the logs that are generated when the webserver reaches out to read the logs?
